shared/terraform.go

```diff
@@ -9,21 +9,32 @@ import (
 	"testing"
 
 	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/rancher/distros-test-framework/config"
 )
 
-func addTerraformOptions(product string) (*terraform.Options, string, error) {
+func setTerraformOptions() (*terraform.Options, string, error) {
+	cfg, err := config.AddEnv()
+	if err != nil {
+		return nil, "", err
+	}
+
 	_, callerFilePath, _, _ := runtime.Caller(0)
 	dir := filepath.Join(filepath.Dir(callerFilePath), "..")
 
 	varDir, err := filepath.Abs(dir +
-		fmt.Sprintf("/config/%s.tfvars", product))
+		fmt.Sprintf("/config/%s.tfvars", cfg.Product))
 	if err != nil {
-		return nil, "", fmt.Errorf("invalid product: %s\n", product)
+		return nil, "", fmt.Errorf("invalid product: %s", cfg.Product)
 	}
 
-	tfDir, err := filepath.Abs(dir + "/modules/" + product)
+	prodOrMod := cfg.Product
```

Comment:
what do you think something more clear as like ?
```
	// we need to validate if we are going to work with airgap module or products.
	// based on that we will set the module name.
	// and also update prod var with the correct product name retrieved from the env.
	var module, prod string
	prod = product
	if airgap != "" {
		module = airgap
	} else {
		module = product
	}
```

Reply:
you can also return the product here now

```
func setTerraformOptions() (*terraform.Options, string, string, error) {

.....
	return terraformOptions, varDir, prod, nil

```

Reply:
Updated